### PR TITLE
refactored test/cmd/templates to use wrapper methods

### PR DESCRIPTION
**Pull request by stevekuznetsov**
/cc @deads2k 

@bparees PTAL

### CODE REVIEW

File: test/cmd/templates.sh [hunk 1]
```diff
@@ -6,58 +6,59 @@ set -o pipefail
...
+os::cmd::expect_success 'oc process ruby-helloworld-sample >/dev/null'
+os::cmd::expect_success 'oc describe templates ruby-helloworld-sample'
+os::cmd::expect_success_and_text 'oc describe templates ruby-helloworld-sample' "BuildConfig.*ruby-sample-build"
+os::cmd::expect_success 'oc delete templates ruby-helloworld-sample'
+os::cmd::expect_success 'oc get templates'
 # TODO: create directly from template
 echo "templates: ok"
 
-oc process -f test/templates/fixtures/guestbook.json -l app=guestbook | oc create -f -
-oc status
```
Comment (stevekuznetsov) [inline test/cmd/templates.sh]: I removed this test because the one right below it subsumes it

File: test/cmd/templates.sh [hunk 2]
```diff
@@ -6,58 +6,59 @@ set -o pipefail
...
-oc process ruby-helloworld-sample >/dev/null
-oc describe templates ruby-helloworld-sample
-[ "$(oc describe templates ruby-helloworld-sample | grep -E "BuildConfig.*ruby-sample-build")" ]
-oc delete templates ruby-helloworld-sample
-oc get templates
+os::cmd::expect_success 'oc get templates'
+os::cmd::expect_success 'oc create -f examples/sample-app/application-template-dockerbuild.json'
+os::cmd::expect_success 'oc get templates'
+os::cmd::expect_success 'oc get templates ruby-helloworld-sample'
+os::cmd::expect_success 'oc get template ruby-helloworld-sample -o json | oc process -f - >/dev/null'
```
Comment (deads2k) [inline test/cmd/templates.sh]: unnecessary redirect

File: test/cmd/templates.sh [hunk 3]
```diff
@@ -6,58 +6,59 @@ set -o pipefail
...
+os::cmd::expect_success 'oc process ruby-helloworld-sample >/dev/null'
```
Comment (deads2k) [inline test/cmd/templates.sh]: unnecessary redirect

File: test/cmd/templates.sh [hunk 4]
```diff
@@ -6,58 +6,59 @@ set -o pipefail
...
+os::cmd::expect_success 'oc describe templates ruby-helloworld-sample'
+os::cmd::expect_success_and_text 'oc describe templates ruby-helloworld-sample' "BuildConfig.*ruby-sample-build"
```
Comment (deads2k) [inline test/cmd/templates.sh]: line 21 is a superset of line 20.

File: test/cmd/templates.sh [hunk 5]
```diff
@@ -6,58 +6,59 @@ set -o pipefail
...
+os::cmd::expect_success_and_text 'oc get pod/template-type-precision -o json' '9223372036854775807'
 # Ensure patch computation preserves data
-oc patch pod template-type-precision -p '{"metadata":{"annotations":{"comment":"patch comment"}}}'
-[ "$(oc get pod/template-type-precision -o json | grep 9223372036854775807)" ]
-[ "$(oc get pod/template-type-precision -o json | grep "patch comment")" ]
-oc delete template/template-type-precision
-oc delete pod/template-type-precision
+patch='{"metadata":{"annotations":{"comment":"patch comment"}}}'
+os::cmd::expect_success "oc patch pod template-type-precision -p '${patch}'"
+os::cmd::expect_success_and_text 'oc get pod/template-type-precision -o json' '922337203685477580'
```
Comment (deads2k) [inline test/cmd/templates.sh]: typo, missing the `7`.

why yes, I am reading these....

File: test/cmd/templates.sh [hunk 6]
```diff
@@ -6,58 +6,59 @@ set -o pipefail
...
-oc process template/ruby-helloworld-sample >/dev/null
-oc process templates/ruby-helloworld-sample > /dev/null
-oc process openshift//ruby-helloworld-sample > /dev/null
-oc process openshift/template/ruby-helloworld-sample >/dev/null
+os::cmd::expect_success 'oc create -f examples/sample-app/application-template-dockerbuild.json -n openshift'
+os::cmd::expect_success 'oc policy add-role-to-user admin test-user'
+os::cmd::expect_success 'oc login -u test-user -p password'
+os::cmd::expect_success 'oc new-project test-template-project'
+os::cmd::expect_success 'oc create -f examples/sample-app/application-template-dockerbuild.json'
+os::cmd::expect_success 'oc process template/ruby-helloworld-sample >/dev/null'
```
Comment (deads2k) [inline test/cmd/templates.sh]: unnecessary redirects here and below

File: test/cmd/templates.sh (hunk 5 above)
Comment (stevekuznetsov) [inline test/cmd/templates.sh]: Good catch

**Comment by stevekuznetsov** [comment]
@deads2k addressed comments

**Comment by deads2k** [comment]
lgtm [merge]

File: test/cmd/templates.sh (hunk 5 above)
Comment (liggitt) [inline test/cmd/templates.sh]: @deads2k code review level-up

**Pull request merged**